crates/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -451,15 +461,21 @@ fn resolve_functions(
     collected_consts: Vec<UnresolvedGlobalConst>,
     self_type: Option<StructId>,
     errors: &mut Vec<CollectedErrors>,
-) -> (Vec<(FileId, FuncId)>, Vec<(FileId, StmtId)>) {
+) -> Vec<(FileId, FuncId)> {
     let mut file_func_ids = Vec::new();
-    let mut global_const_ids = Vec::new();
+
     // Lower each function in the crate. This is now possible since imports have been resolved
     for unresolved_functions in collected_functions {
         let file_id = unresolved_functions.file_id;
         let mut collected_errors = CollectedErrors { file_id, errors: Vec::new() };
 
-        for (mod_id, func_id, func) in unresolved_functions.functions {
+        for (mod_id, func_id, mut func) in unresolved_functions.functions {
+            let mut body = func.clone().def.body.0;
+            for global_const in collected_consts.clone() {
+                body.insert(0, global_const.stmt_def);
```

Comment:
I'm still not too keen on just inserting these declarations into every function. Does name resolution with globals declared in submodules still work as expected?
This is more of a note for later if this PR is finalized but it looks like only the body needs to be cloned and the inserting can be made more efficient by appending the body to the collected_consts clone instead of doing repeated inserts.

Reply:
I am not keen on it as well. I tested out globals declared in submodules and as you suspected it does not work, while using the strategy in PR #267 it does work. The other comments are noted as well though.